data/publiccloud/terraform/azure.tf

```diff
@@ -197,6 +201,8 @@ resource "azurerm_linux_virtual_machine" "openqa-vm" {
   timeouts {
     create = var.vm_create_timeout
   }
+
+  user_data = var.cloud_init != "" ? filebase64(var.cloud_init) : "Cg=="
```

Comment:
I recently learn about `dynamic` terraform pattern

```
  dynamic "user_data" {
    for_each = var.cloud_init ? [filebase64(var.cloud_init)] : []

    content {
      user_data = each
    }
  }
 ```

Reply:
```
Error: Unsupported block type

  on plan.tf line 205, in resource "azurerm_linux_virtual_machine" "openqa-vm":
 205:   dynamic "user_data" {

Blocks of type "user_data" are not expected here.
```

Reply:
I think that your suggestion gets expanded to:
```
user_data {
  user_data = "XYZ==="
}
```

And I don't have clue how to fix it.

Reply:
what does "Cg==" means ? :thinking:  I think some short comment explaining this might be useful

Reply:
The `Cg==` is empty string encoded in base64. I replaced it with `null` today. Please recheck.